Cannot use ansible filters with ansible-galaxy

<!--- Verify first that your issue is not already reported on GitHub -->
<!--- Also test if the latest release and devel branch are affected too -->
<!--- Complete *all* sections as described, this form is processed automatically -->

##### SUMMARY
<!--- Explain the problem briefly below -->
I am trying to use ansible filters to template some files in my role skeleton, but ansible-galaxy cannot find them. Only jinja2 filters work.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below, use your best guess if unsure -->
ansible-galaxy

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
ansible 2.10.0.dev0
  config file = None
  configured module search path = ['/home/nikos/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/nikos/.local/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.1 (default, Jan 22 2020, 06:38:00) [GCC 9.2.0]
```

##### CONFIGURATION
<!--- Paste verbatim output from "ansible-config dump --only-changed" between quotes -->
```paste below
none
```

##### OS / ENVIRONMENT
<!--- Provide all relevant information below, e.g. target OS versions, network device firmware, etc. -->


##### STEPS TO REPRODUCE
<!--- Describe exactly how to reproduce the problem, using a minimal test-case -->

Create a file like the one below in your role skeleton:

<!--- Paste example playbooks or commands between quotes below -->
```yaml
{{ role_name | replace('-', '_') }}_flags:
  - name: {{ role_name | replace('-', '_') }}_flag
    id: {{ role_name | to_uuid | truncate(6, True, '', 0) }}
...
```

<!--- HINT: You can paste gist.github.com links for larger files -->

##### EXPECTED RESULTS
<!--- Describe what you expected to happen when running the steps above -->
I would expect `to_uuid` to run fine.

##### ACTUAL RESULTS
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->

<!--- Paste verbatim command output between quotes -->
```paste below
ERROR! Unexpected Exception, this is probably a bug: no filter named 'to_uuid'
the full traceback was:

Traceback (most recent call last):
  File "/usr/bin/ansible-galaxy", line 123, in <module>
    exit_code = cli.run()
  File "/home/nikos/.local/lib/python3.8/site-packages/ansible/cli/galaxy.py", line 479, in run
    context.CLIARGS['func']()
  File "/home/nikos/.local/lib/python3.8/site-packages/ansible/cli/galaxy.py", line 882, in execute_init
    template_env.get_template(src_template).stream(inject_data).dump(dest_file, encoding='utf-8')
  File "/usr/lib/python3.8/site-packages/jinja2/environment.py", line 883, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/usr/lib/python3.8/site-packages/jinja2/environment.py", line 857, in _load_template
    template = self.loader.load(self, name, globals)
  File "/usr/lib/python3.8/site-packages/jinja2/loaders.py", line 129, in load
    code = environment.compile(source, name, filename)
  File "/usr/lib/python3.8/site-packages/jinja2/environment.py", line 638, in compile
    self.handle_exception(source=source_hint)
  File "/usr/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/usr/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/nikos/Projects/ethical-hacking/EN2720/resources/skeletons/vulnerability-role/defaults/flags.yml.j2", line 8, in template
    id: {{ role_name | to_uuid | truncate(6, True, '', 0) }}
jinja2.exceptions.TemplateAssertionError: no filter named 'to_uuid'
```

`replace` runs fine.

--- a/changelogs/fragments/69104-galaxy-cli-templar.yml
+++ b/changelogs/fragments/69104-galaxy-cli-templar.yml
@@ -1,0 +1,4 @@
+bugfixes:
+- ansible-galaxy - Utilize ``Templar`` for templating skeleton files, so that
+  they have access to Ansible filters/tests/lookups
+  (https://github.com/ansible/ansible/issues/69104)

--- a/lib/ansible/cli/galaxy.py
+++ b/lib/ansible/cli/galaxy.py
@@ -12,7 +12,6 @@ import textwrap
 import time
 import yaml
 
-from jinja2 import BaseLoader, Environment, FileSystemLoader
 from yaml.error import YAMLError
 
 import ansible.constants as C
@@ -40,8 +39,10 @@ from ansible.module_utils.ansible_release import __version__ as ansible_version
 from ansible.module_utils.common.collections import is_iterable
 from ansible.module_utils._text import to_bytes, to_native, to_text
 from ansible.module_utils import six
+from ansible.parsing.dataloader import DataLoader
 from ansible.parsing.yaml.loader import AnsibleLoader
 from ansible.playbook.role.requirement import RoleRequirement
+from ansible.template import Templar
 from ansible.utils.display import Display
 from ansible.utils.plugin_docs import get_versioned_doclink
 
@@ -667,15 +668,11 @@ class GalaxyCLI(CLI):
 
             return textwrap.fill(v, width=117, initial_indent="# ", subsequent_indent="# ", break_on_hyphens=False)
 
-        def to_yaml(v):
-            return yaml.safe_dump(v, default_flow_style=False).rstrip()
+        loader = DataLoader()
+        templar = Templar(loader, variables={'required_config': required_config, 'optional_config': optional_config})
+        templar.environment.filters['comment_ify'] = comment_ify
 
-        env = Environment(loader=BaseLoader)
-        env.filters['comment_ify'] = comment_ify
-        env.filters['to_yaml'] = to_yaml
-
-        template = env.from_string(meta_template)
-        meta_value = template.render({'required_config': required_config, 'optional_config': optional_config})
+        meta_value = templar.template(meta_template)
 
         return meta_value
 
@@ -790,6 +787,7 @@ class GalaxyCLI(CLI):
                 documentation_url='http://docs.example.com',
                 homepage_url='http://example.com',
                 min_ansible_version=ansible_version[:3],  # x.y
+                dependencies=[],
             ))
 
             obj_path = os.path.join(init_path, obj_name)
@@ -839,7 +837,8 @@ class GalaxyCLI(CLI):
                 to_native(obj_skeleton), galaxy_type)
             )
 
-        template_env = Environment(loader=FileSystemLoader(obj_skeleton))
+        loader = DataLoader()
+        templar = Templar(loader, variables=inject_data)
 
         # create role directory
         if not os.path.exists(b_obj_path):
@@ -877,9 +876,12 @@ class GalaxyCLI(CLI):
                     with open(b_dest_file, 'wb') as galaxy_obj:
                         galaxy_obj.write(to_bytes(meta_value, errors='surrogate_or_strict'))
                 elif ext == ".j2" and not in_templates_dir:
-                    src_template = os.path.join(rel_root, f)
+                    src_template = os.path.join(root, f)
                     dest_file = os.path.join(obj_path, rel_root, filename)
-                    template_env.get_template(src_template).stream(inject_data).dump(dest_file, encoding='utf-8')
+                    template_data = to_text(loader._get_file_contents(src_template)[0], errors='surrogate_or_strict')
+                    b_rendered = to_bytes(templar.template(template_data), errors='surrogate_or_strict')
+                    with open(dest_file, 'wb') as df:
+                        df.write(b_rendered)
                 else:
                     f_rel_path = os.path.relpath(os.path.join(root, f), obj_skeleton)
                     shutil.copyfile(os.path.join(root, f), os.path.join(obj_path, f_rel_path))

--- a/lib/ansible/galaxy/data/default/collection/galaxy.yml.j2
+++ b/lib/ansible/galaxy/data/default/collection/galaxy.yml.j2
@@ -1,11 +1,11 @@
 ### REQUIRED
 {% for option in required_config %}
 {{ option.description | comment_ify }}
-{{ {option.key: option.value} | to_yaml }}
+{{ {option.key: option.value} | to_nice_yaml }}
 {% endfor %}
 
 ### OPTIONAL but strongly recommended
 {% for option in optional_config %}
 {{ option.description | comment_ify }}
-{{ {option.key: option.value} | to_yaml }}
+{{ {option.key: option.value} | to_nice_yaml }}
 {% endfor %}

--- a/test/units/cli/test_galaxy.py
+++ b/test/units/cli/test_galaxy.py
@@ -728,7 +728,7 @@ def test_collection_build(collection_artifact):
             elif file_entry['name'] == 'README.md':
                 assert file_entry['ftype'] == 'file'
                 assert file_entry['chksum_type'] == 'sha256'
-                assert file_entry['chksum_sha256'] == '45923ca2ece0e8ce31d29e5df9d8b649fe55e2f5b5b61c9724d7cc187bd6ad4a'
+                assert file_entry['chksum_sha256'] == '6d8b5f9b5d53d346a8cd7638a0ec26e75e8d9773d952162779a49d25da6ef4f5'
             else:
                 assert file_entry['ftype'] == 'dir'
                 assert file_entry['chksum_type'] is None

